cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -499,54 +499,76 @@ metadata property or metadata tag, submit an HTTP GET request::
-                 "MapReduce:PurchaseHistoryBuilder":"PurchaseHistoryBuilder"
-              },
-              "tags":[
-                 "Purchase",
-                 "PurchaseHistory"
-              ]
-           }
-        }
-     },
-     {
-        "entityId":{
-           "id":{
-              "instanceId":"history",
-              "namespace":{
-                 "id":"default"
-              }
-           },
-           "type":"datasetinstance"
+  {
+    "cursors": [ ],
+    "limit": 20,
+    "numCursors": 0,
+    "offset": 0,
+    "showHidden": false,
+    "sort": "creation-time DESC",
+    "total": 2,
+    "results": [
+        {
+            "entityId": {
+                "id": {
+                    "application": {
+                        "applicationId": "WordCount",
+                        "namespace": {
+                            "id": "default"
+                        }
+                    },
+                    "id": "RetrieveCounts",
+                    "type": "Service"
+                },
+                "type": "program"
+            },
+            "metadata": {
+                "SYSTEM": {
+                    "properties": {
+                        "creation-time": "1482091087438",
+                        "description": "A service to retrieve statistics, word counts, and associations.",
+                        "entity-name": "RetrieveCounts",
+                        "version": "-SNAPSHOT"
+                    },
+                    "tags": [
+                        "Realtime",
+                        "Service"
+                    ]
+                }
+            }
         },
-        "metadata":{
-           "SYSTEM":{
-              "properties":{
-                 "type":"co.cask.cdap.examples.purchase.PurchaseHistoryStore"
-              },
-              "tags":[
-                 "history",
-                 "explore",
-                 "batch"
-              ]
-           }
+        {
+            "entityId": {
+                "id": {
+                    "application": {
+                        "applicationId": "WordCount",
+                        "namespace": {
+                            "id": "default"
+                        }
+                    },
+                    "id": "WordCounter",
+                    "type": "Flow"
+                },
+                "type": "program"
+            },
+            "metadata": {
+                "SYSTEM": {
+                    "properties": {
+                        "creation-time": "1482091087390",
+                        "description": "Example Word Count Flow",
+                        "entity-name": "WordCounter",
+                        "version": "-SNAPSHOT"
+                    },
+                    "tags": [
+                        "Flow",
+                        "Realtime"
+                    ]
+                }
+            }
         }
-     }
-  ]
+    ]
+  }
+
 
 .. list-table::
```

Comment:
This parameter table has gotten quite long, and should now be moved up and precede the returned results, as we do for other APIs. It has become too far from the command to even check here.

Also, I think that the order of the parameters should change; I would suggest:
``namespace-id``
``entity-type``
``term``
then followed by the options, in alphabetical order.

I think the command should become:
```
GET /v3/namespaces/<namespace-id>/metadata/search?query=<term>[&target=<entity-type>&target=<entity-type2>...][&cursor=<cursor>][&limit=<limit>][&numCursors=<num-cursors>][&offset=<offset>][&showHidden=false][&sort=<field-name>+<order>]